File: ATM_INTERFACE.java
import java.util.Scanner;

class General_Bank_Account {
    private double balance;

    public General_Bank_Account(double initialBalance) {
        this.balance = initialBalance;
    }

    public double getBalance() {
        return balance;
    }

    public void deposit(double amount) {
        balance += amount;
    }

    public boolean withdraw(double amount) {
        if (amount <= balance) {
            balance -= amount;
            return true;
        } else {
            System.out.println("Fund is not sufficient!");
            return false;
        }
    }
}

class ATM {
    private General_Bank_Account bankAccount;
    private Scanner scanner;

    public ATM(General_Bank_Account bankAccount) {
        this.bankAccount = bankAccount;
        this.scanner = new Scanner(System.in);
    }

    public void displayMenu() {
        System.out.println("1. Withdraw");
        System.out.println("2. Deposit");
        System.out.println("3. Check Balance");
        System.out.println("4. Exit");
    }

    public void handleOption(int option) {
        switch (option) {
            case 1:
                System.out.print("Enter withdrawal amount: ");
                double withdrawAmount = scanner.nextDouble();
                if (withdrawAmount > 0) {
                    bankAccount.withdraw(withdrawAmount);
                } else {
                    System.out.println("Amount you have entered is not valid !");
                }
                break;

            case 2:
                System.out.print("Enter deposit amount: ");
                double depositAmount = scanner.nextDouble();
                if (depositAmount > 0) {
                    bankAccount.deposit(depositAmount);
                } else {
                    System.out.println("Amount is not valid !");
                }
                break;

            case 3:
                System.out.println("Current Balance in you bank account: $" + bankAccount.getBalance());
                break;

            case 4:
                System.out.println("Exiting ATM. Thank you !");
                System.exit(0);

            default:
                System.out.println("Option is not valid !");
        }
    }
}

public class ATM_INTERFACE {
    public static void main(String[] args) {
        General_Bank_Account userAccount = new General_Bank_Account(1000.0); // Initial balance
        ATM atm = new ATM(userAccount);

        while (true) {
            atm.displayMenu();
            System.out.print("Enter your choice: ");
            int choice = new Scanner(System.in).nextInt();
            atm.handleOption(choice);
        }
    }
}

File: ATM_INTERFACE.py
class GeneralBankAccount:
    def __init__(self, initial_balance):
        self.balance = initial_balance

    def get_balance(self):
        return self.balance

    def deposit(self, amount):
        self.balance += amount
        print(f"${amount} deposited successfully.")

    def withdraw(self, amount):
        if amount <= self.balance:
            self.balance -= amount
            print(f"${amount} withdrawn successfully.")
            return True
        else:
            print("Fund is not sufficient!")
            return False


class ATM:
    def __init__(self, bank_account):
        self.bank_account = bank_account

    def display_menu(self):
        print("\n===== ATM MENU =====")
        print("1. Withdraw")
        print("2. Deposit")
        print("3. Check Balance")
        print("4. Exit")

    def handle_option(self, option):
        if option == 1:
            withdraw_amount = float(input("Enter withdrawal amount: "))
            if withdraw_amount > 0:
                self.bank_account.withdraw(withdraw_amount)
            else:
                print("Amount entered is not valid!")

        elif option == 2:
            deposit_amount = float(input("Enter deposit amount: "))
            if deposit_amount > 0:
                self.bank_account.deposit(deposit_amount)
            else:
                print("Amount entered is not valid!")

        elif option == 3:
            print(f"Current Balance: ${self.bank_account.get_balance()}")

        elif option == 4:
            print("Exiting ATM. Thank you!")
            exit()

        else:
            print("Option is not valid!")


# Main Program
if __name__ == "__main__":
    user_account = GeneralBankAccount(1000.0)  # Initial balance
    atm = ATM(user_account)

    while True:
        atm.display_menu()
        try:
            choice = int(input("Enter your choice: "))
            atm.handle_option(choice)
        except ValueError:
            print("Please enter a valid number!")
